From 86c22ea560a0b2853bef25653db0ef4f6b816e7d Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Mon, 13 Nov 2023 20:15:39 +0800
Subject: [PATCH 04/17] mtk: wifi: mt76: mt7996: set RCPI value in rate control
 command

Set RCPI values in mt7996_mcu_sta_rate_ctrl_tlv(), which can make the
FW rate control be initialized with a better MCS selection table.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/mcu.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -1977,6 +1977,7 @@ static void
 mt7996_mcu_sta_rate_ctrl_tlv(struct sk_buff *skb, struct mt7996_dev *dev,
 			     struct ieee80211_vif *vif, struct ieee80211_sta *sta)
 {
+#define INIT_RCPI 180
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt76_phy *mphy = mvif->phy->mt76;
 	struct cfg80211_chan_def *chandef = &mphy->chandef;
@@ -2074,6 +2075,8 @@ mt7996_mcu_sta_rate_ctrl_tlv(struct sk_b
 					       IEEE80211_HE_6GHZ_CAP_MAX_AMPDU_LEN_EXP);
 	}
 	ra->sta_cap = cpu_to_le32(cap);
+
+	memset(ra->rx_rcpi, INIT_RCPI, sizeof(ra->rx_rcpi));
 }
 
 int mt7996_mcu_add_rate_ctrl(struct mt7996_dev *dev, struct ieee80211_vif *vif,
